app/controllers/certificates_controller.rb

```diff
@@ -14,14 +14,25 @@ def create
     end
   end
 
+  def update
+    @certificate = Certificate.find(params[:id])
+    @certificate.update_attributes(update_params)
+    redirect_to component_path(@certificate.component_id)
+  end
+
   private
 
   def upload_params
     component_id ||= params[:component_id]
 
     params.require(:certificate)
-          .permit(:value,:usage)
+          .permit(:value, :usage, :id)
```

Comment:
Any specific reason this was added? The `id` should be `nil` when we're uploading and there shouldn't be a parameter.

Reply:
Yup, I think I was playing around with what this did and didn't reverse it all out. I'll sort that.